### PR TITLE
upgrade package and config to use CORS

### PR DESCRIPTION
**Pull request by ewitkon**
This PR update the package to the latest version and fix minor issues in code.
This PR also update the configuration to use localhost CORS policy if possible.

### CODE REVIEW

File: .env.development.sample [hunk 1]
```diff
@@ -1,4 +1,10 @@
...
+FOUNDRY_API_URL=https://<YOUR FOUNDRY STACK DOMAIN>
+APPLICATION_REDIRECT_URL=http://localhost:8080/auth/callback
 FOUNDRY_CLIENT_ID=<Fill in Client ID for your application>
-API_PROXY_TARGET_URL=<Fill in your Foundry instance base URL, e.g. https://example.palantir.com, for development purposes>
+
+# If you did not add http://localhost:8080 to your CORS policy in Control panel comment above and uncomment below settings
+# FOUNDRY_API_URL=https://localhost:8080
+# APPLICATION_REDIRECT_URL=https://localhost:8080/auth/callback
+# FOUNDRY_CLIENT_ID=<Fill in Client ID for your application>
+# API_PROXY_TARGET_URL=<Fill in your Foundry instance base URL, e.g. https://example.palantir.com, for development purposes>
```
Comment (styu) [inline .env.development.sample:10]: i think i'd prefer if this lived in a `.env.development.no-cors-sample` file 🤔

File: README.md [hunk 2]
```diff
@@ -12,34 +12,28 @@ A simple skeleton application to get you started on React/TypeScript development
 1. Copy or make a fork of this repository (you can delete `.policy.yml`)
 1. Run `npm install`
 1. Create an application via the Foundry Developer Console with the configuration below. Once the application has been created, come back to these instructions.
-    1. Select client facing application (creates a public client) when asked what type of application you're building
-    1. Add `https://localhost:8080/auth/callback` as a redirect URL when asked for one
-    1. If you know where your application is getting deployed, you can add your production URL as well: `https://example.app.com/auth/callback`
+    1. Add `http://localhost:8080` to your CORS policy in Control Panel, see [Configure CORS](https://www.palantir.com/docs/foundry/administration/configure-cors/index.html) for more details.
```
Comment (styu) [inline README.md:15]:
```suggestion
    1. Follow the instructions in [Configure CORS](https://www.palantir.com/docs/foundry/administration/configure-cors/index.html) to add  `http://localhost:8080` to your CORS policy in Control Panel.
```

File: README.md [hunk 3]
```diff
@@ -12,34 +12,28 @@ A simple skeleton application to get you started on React/TypeScript development
...
+    2. Select client facing application (creates a public client) when asked what type of application you're building
```
Comment (styu) [inline README.md:16]:
```suggestion
    2. Select **client facing application** (creates a public client) when asked what type of application you're building. Creating a confidential client means you **cannot** login through the browser, and this cannot be changed after creating the app.
```

File: README.md [hunk 4]
```diff
@@ -12,34 +12,28 @@ A simple skeleton application to get you started on React/TypeScript development
...
+    3. Add `http://localhost:8080/auth/callback` as a redirect URL when asked for one
+    4. If you know where your application is getting deployed, you can add your production URL as well: `https://example.app.com/auth/callback`
 1. Follow the instructions in the Foundry Developer Console to generate and then install your application's Ontology SDK via the application-specific NPM registry
 1. Run `mv .env.development.sample .env.development` to start setting up your environment variables, and fill in the fields marked with `<>`
 
-    a. You can find your Client ID in either the Quickstart guide for the SDK or in the Permissions & OAuth page
+    Note! You can find your Client ID in the Permissions & OAuth page or in the Getting started guid under the API documentation
```
Comment (styu) [inline README.md:22]:
```suggestion
    > Note: You can find your Client ID in the **Permissions & OAuth** page or in the **Getting started guide** inside your application's API documentation.
```

File: README.md [hunk 5]
```diff
@@ -12,34 +12,28 @@ A simple skeleton application to get you started on React/TypeScript development
...
 
 1. Update the following files with your Ontology SDK and Object types:
     1. Update [`src/utils/client.ts`](./src/utils/client.ts) with the correct package name
-    1. Update [`src/pages/home.tsx`](./src/pages/home.tsx) with the correct package name and Ontology Objects you want to use.
+    2. Update [`src/pages/home.tsx`](./src/pages/home.tsx) with the correct package name and Ontology Objects you want to use.
```
Comment (styu) [inline README.md:26]: github automaticaly numbers these so we don't have to keep them up to date manually
```suggestion
    1. Update [`src/pages/home.tsx`](./src/pages/home.tsx) with the correct package name and Ontology Objects you want to use.
```

File: webpack.config.js [hunk 6]
```diff
@@ -92,15 +94,17 @@ module.exports = {
...
+            ? []
+            : [
+                  {
+                      // This proxies calls from the browser to the configured Foundry instance
+                      target: process.env.API_PROXY_TARGET_URL,
+                      context: ["/multipass/api/**", "/api/**"],
+                      changeOrigin: true,
+                      secure: true,
+                  },
+              ],
```
Comment (styu) [inline webpack.config.js:108]: i think i'd rather keep this clean and remove it, and move this into the FAQ in the README instead.

File: .env.development.no-cors.sample [hunk 7]
```diff
@@ -0,0 +1,3 @@
+FOUNDRY_API_URL=https://<YOUR FOUNDRY STACK DOMAIN>
```
Comment (styu) [inline .env.development.no-cors.sample:1]: i meant something a bit different - if we wanted to keep the old env variables i was suggesting having them in a file called `.env.development.no-cors.sample`. this one (the default) should still be called `.env.development.sample`

Comment (styu) [reply .env.development.no-cors.sample:1]: (so basically swap the names)

File: README.md [hunk 8]
```diff
@@ -56,6 +50,31 @@ During local development API requests are routed through a [webpack-dev-server p
...
+                    // This proxies calls from the browser to the configured Foundry instance
+                    target: process.env.API_PROXY_TARGET_URL,
+                    context: ["/multipass/api/**", "/api/**"],
+                    changeOrigin: true,
+                    secure: true,
+                },
+            ],
+```
+
+4. Test your application by running `npm run dev` and visiting https://localhost:8080 (note HTTPS and not HTTP)
```
Comment (styu) [inline README.md:80]: might be worth leaving a note saying that you'll get a this is unsafe error, and that is expected

Comment (ewitkon) [reply README.md:80]: Done!

File: .gitignore [hunk 9]
```diff
@@ -1,5 +1,5 @@
 .DS_Store
-.env*
+.env.development
```
Comment (styu) [inline .gitignore:2]:
```suggestion
.env.development
.env.production
```

File: README.md [hunk 10]
```diff
@@ -12,34 +12,28 @@ A simple skeleton application to get you started on React/TypeScript development
 1. Copy or make a fork of this repository (you can delete `.policy.yml`)
 1. Run `npm install`
 1. Create an application via the Foundry Developer Console with the configuration below. Once the application has been created, come back to these instructions.
-    1. Select client facing application (creates a public client) when asked what type of application you're building
-    1. Add `https://localhost:8080/auth/callback` as a redirect URL when asked for one
-    1. If you know where your application is getting deployed, you can add your production URL as well: `https://example.app.com/auth/callback`
+    1. Follow the instructions in [Configure CORS](https://www.palantir.com/docs/foundry/administration/configure-cors/index.html) to add  `http://localhost:8080` to your CORS policy in Control Panel.
+    2. Select **client facing application** (creates a public client) when asked what type of application you're building. Creating a confidential client means you **cannot** login through the browser, and this cannot be changed after creating the app.
```
Comment (styu) [inline README.md:16]:
```suggestion
    2. Select **Client facing application** (creates a public client) when asked what type of application you're building. Creating a confidential client means you **cannot** login through the browser, and this cannot be changed after creating the app.
```

File: README.md [hunk 11]
```diff
@@ -56,6 +50,32 @@ During local development API requests are routed through a [webpack-dev-server p
...
 export NODE_EXTRA_CA_CERTS=/path/to/your/cert.pem
 ```
+
+### I do not have permission to add `http://localhost:8080` to my CORS policy in control panel
+
+If you do not have permission to do so and can't get Foundry Admin to approve it you can use the following setup:
+
+1. Add `https://localhost:8080/auth/callback` as a redirect URL in the Permission & OAuth tab on the left menu.
+1. Run `mv .env.development.no-core.sample .env.development` to start setting up your environment variables, and fill in the fields marked with `<>`
+1. Add the following section to your `devServer: ` section in the webpack.config.js file, after `port: 8080,`
```
Comment (styu) [inline README.md:63]:
```suggestion
1. Add the following section to your `devServer` section in the webpack.config.js file:
```

**Review by styu — APPROVED** [review]

**Pull request merged**